wasm/sexpressions.py

```diff
@@ -0,0 +1,232 @@
+  <val_type>.<cvtop>_<val_type>(_<sign>)?
+
+func:    ( func <name>? <func_type> <local>* <instr>* )
+         ( func <name>? ( export <string> ) <...> )                         ;; = (export <string> (func <N>)) (func <name>? <...>)
+         ( func <name>? ( import <string> <string> ) <func_type>)           ;; = (import <name>? <string> <string> (func <func_type>))
+param:   ( param <val_type>* ) | ( param <name> <val_type> )
+result:  ( result <val_type>* )
+local:   ( local <val_type>* ) | ( local <name> <val_type> )
+
+global:  ( global <name>? <global_type> <instr>* )
+         ( global <name>? ( export <string> ) <...> )                       ;; = (export <string> (global <N>)) (global <name>? <...>)
+         ( global <name>? ( import <string> <string> ) <global_type> )      ;; = (import <name>? <string> <string> (global <global_type>))
+table:   ( table <name>? <table_type> )
+         ( table <name>? ( export <string> ) <...> )                        ;; = (export <string> (table <N>)) (table <name>? <...>)
+         ( table <name>? ( import <string> <string> ) <table_type> )        ;; = (import <name>? <string> <string> (table <table_type>))
+         ( table <name>? ( export <string> )* <elem_type> ( elem <var>* ) ) ;; = (table <name>? ( export <string> )* <size> <size> <elem_type>) (elem (i32.const 0) <var>*)
+elem:    ( elem <var>? (offset <instr>* ) <var>* )
+         ( elem <var>? <expr> <var>* )                                      ;; = (elem <var>? (offset <expr>) <var>*)
+memory:  ( memory <name>? <memory_type> )
+         ( memory <name>? ( export <string> ) <...> )                       ;; = (export <string> (memory <N>))+ (memory <name>? <...>)
+         ( memory <name>? ( import <string> <string> ) <memory_type> )      ;; = (import <name>? <string> <string> (memory <memory_type>))
+         ( memory <name>? ( export <string> )* ( data <string>* ) )         ;; = (memory <name>? ( export <string> )* <size> <size>) (data (i32.const 0) <string>*)
+data:    ( data <var>? ( offset <instr>* ) <string>* )
+         ( data <var>? <expr> <string>* )                                   ;; = (data <var>? (offset <expr>) <string>*)
+
+start:   ( start <var> )
+
+typedef: ( type <name>? ( func <param>* <result>* ) )
+
+import:  ( import <string> <string> <imkind> )
+imkind:  ( func <name>? <func_type> )
+         ( global <name>? <global_type> )
+         ( table <name>? <table_type> )
+         ( memory <name>? <memory_type> )
+export:  ( export <string> <exkind> )
+exkind:  ( func <var> )
+         ( global <var> )
+         ( table <var> )
+         ( memory <var> )
+
+module:  ( module <name>? <typedef>* <func>* <import>* <export>* <table>? <memory>? <global>* <elem>* <data>* <start>? )
+         <typedef>* <func>* <import>* <export>* <table>? <memory>? <global>* <elem>* <data>* <start>?  ;; =
+         ( module <typedef>* <func>* <import>* <export>* <table>? <memory>? <global>* <elem>* <data>* <start>? )
+"""
+from pyparsing import (
```

Comment:
We're already using `parsimonious` in `eth-abi` which is a little more user friendly IMHO.  Any thoughts about using that?  The nice thing about parsimonious is that you can define grammars using production-rule-like syntax.

Reply:
`pyparsing` was just what turned up when I was digging for reference implementations of how to parse S-Expressions.  :+1: to `parsimonious` if you like it better.